include/vcmi/Entity.h

```diff
@@ -14,6 +14,12 @@ VCMI_LIB_NAMESPACE_BEGIN
 
 class IBonusBearer;
 
+class DLL_LINKAGE WithBonuses
+{
+public:
+	virtual const IBonusBearer * accessBonuses() const = 0;
```

Comment:
maybe `toBonusBearer()` would be a better name?

Reply:
@DjWarmonger then we should redesign all entities to it. Should I do it now?

Reply:
What is actually the purpose of this class? Just the wrapper of `IBonusBearer`?

Reply:
Yes, and this wrapper is extensively used in Lua, and I think it is good to use it anywhere where we do not need write access to bonus tree.

Reply:
Okay, I still think we shoudl use as few names and conventions as possible. Maybe just `getBonusBearer()`.

Reply:
There was an EntityWithBonuses class, and I splitted "WithBonuses" part from it.

Reply:
If there is no other scripts than use accessBonuses exists, then I can rename it to getBonusBearer. Anyway, this class is 6 years old)

Reply:
@IvanSavenko @DjWarmonger there is other similar method in BattleInfo, called asBearer. Maybe we should choose a common name for it, if we need to change?

Reply:
And getBattleNode exists too for a same purpose. I think it is a good refactoring to unify it under a same interface.

Reply:
`toBonusBearer` might be slightly better since this is effectively type casting. But not against `getBonusBearer`. Both are better than current `accessBonuses`.

Considering that we have basically zero scripts rename of a method related to scripts it not a problem

Reply:
Or perhaps getBonusBearer after all? If only for consistency.

toBonusBearer might better represent method since this is type conversion and not getter. On the other hand we can view this fact internal info and provide `getBonusBearer` everywhere

Reply:
Maybe use `toBearer`? Same semantic, but less words.

Reply:
@IvanSavenko I agree with you and think than outside classes should not know about is class a bonus bearer or not.

Reply:
@IvanSavenko @DjWarmonger what name is better? `getBonusBearer`, `getBearer`, `toBonusBearer`, `toBearer`?

Reply:
My preference:
1) getBonusBearer
2) toBonusBearer

Reply:
So, longer is better. Will do getBonusBearer then.

Reply:
All changes pushed.